crates/ruff_linter/src/rules/ruff/rules/parenthesize_logical_operators.rs

```diff
@@ -0,0 +1,82 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast as ast;
+use ruff_python_ast::parenthesize::parenthesized_range;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for chained operators where adding parentheses could improve the
+/// clarity of the code.
+///
+/// ## Why is this bad?
+/// `and` always binds more tightly than `or` when chaining the two together,
+/// but this can be hard to remember (and sometimes surprising).
+/// Adding parentheses in these situations can greatly improve code readability,
+/// with no change to semantics or performance.
+///
+/// For example:
+/// ```python
+/// a, b, c = 1, 0, 2
+/// x = a or b and c
+///
+/// d, e, f = 0, 1, 2
+/// y = d and e or f
+/// ```
+///
+/// Use instead:
+/// ```python
+/// a, b, c = 1, 0, 2
+/// x = a or (b and c)
+///
+/// d, e, f = 0, 1, 2
+/// y = (d and e) or f
+/// ````
+#[violation]
+pub struct ParenthesizeChainedOperators;
+
+impl Violation for ParenthesizeChainedOperators {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear"
```

Comment:
Thoughts on the error message here? I was torn between the current one and this, which is more verbose but maybe easier to understand:

```suggestion
            "Parenthesize `a and b` expressions when chaining `and` and `or` together, to make it clear that `and` binds more tightly than `or`"
```

Reply:
I have a minor pref for the version you already have here. We tend towards having shorter messages, and put motivation in the documentation... In the future, we'll also include links to the documentation in the linter output.

Reply:
Agree with the shorter one too. I would expect most programmers to know what "precedence" means.

Reply:
Yeah, I also much prefer to keep things concise where possible.

> In the future, we'll also include links to the documentation in the linter output.

That sounds awesome!